### DISCUSSION TITLE
rust-analyzer support

### DISCUSSION THREAD

**Discussion by zhiqiangxu** [hverlin/mise-vscode#70] [General]
<img width="1288" alt="image" src="https://github.com/user-attachments/assets/81292f0b-4b93-4bd1-9886-177c87d73455" />

However if I type `which cargo` directly, it can be found:

```bash
% which cargo
/Users/alanxu/.cargo/bin/cargo
```

Here is my `settings.json` automatically generated by mise-vscode extension:

<img width="628" alt="image" src="https://github.com/user-attachments/assets/aef131af-99fb-489d-895f-ada292cb5f72" />

**Comment by hverlin** [comment]
Thanks for the report. Please note that currently, `mise-vscode` does not try to configure any settings regarding the rust-analyzer extension. ([list of currently supported extensions](https://hverlin.github.io/mise-vscode/reference/supported-extensions/))

I have managed to reproduce your issue. This happens if you have installed rust (e.g., with [mise](https://mise.jdx.dev/lang/rust.html)) but did not add the cargo bin to `PATH`.

Unfortunately, there is no way to cleanly override the path to cargo for `rust-analyzer` (see https://github.com/rust-lang/rust-analyzer/issues/16204). I think I can't do much apart from documenting a workaround.

## Workarounds
### Add shims to PATH
Follow the recommended approach here: https://mise.jdx.dev/ide-integration.html#adding-shims-to-path-default-shell

For example, if you use `bash`, you can add the following to your `~/.profile`.
```sh
eval "$(mise activate bash --shims)"
```

### Add cargo bin to path
You can also  update your shell rc file with:
```sh
export PATH="$HOME/.cargo/bin:$PATH"
```
then re-launch VSCode

**Reply by zhiqiangxu** [reply]
The `Add cargo bin to path` workaround works! I'm not sure what's the difference between `mise activate bash --shims` and  `mise activate bash`? The former only adds a path to the `$PATH` variable, but the later outputs a lot of functions.
